cmd/create/accountroles/cmd.go

```diff
@@ -358,120 +360,144 @@ func run(cmd *cobra.Command, argv []string) {
-			Build()
+		assumeRolePolicy := getAssumeRolePolicy(file, input)
 
-		policyName := aws.GetPolicyName(accRoleName)
-		createPolicy := awscb.NewIAMCommandBuilder().
-			SetCommand(awscb.CreatePolicy).
-			AddParam(awscb.PolicyName, policyName).
-			AddParam(awscb.PolicyDocument, fmt.Sprintf("file://sts_%s_permission_policy.json", file)).
-			AddTags(iamTags).
-			AddParam(awscb.Path, path).
-			Build()
+		r.Reporter.Debugf("Creating role '%s'", accRoleName)
+		tagsList := mp.getRoleTags(file, input)
+		roleARN, err := r.AWSClient.EnsureRole(accRoleName, assumeRolePolicy, input.permissionsBoundary,
+			input.defaultPolicyVersion, tagsList, input.path, true)
+		if err != nil {
+			return err
+		}
+		r.Reporter.Infof("Created role '%s' with ARN '%s'", accRoleName, roleARN)
 
-		// Determine policy ARN
+		filename := fmt.Sprintf("sts_%s_permission_policy", file)
 		var policyARN string
-		var err error
-		if managedPolicies {
-			policyARN, err = aws.GetManagedPolicyARN(policies, fmt.Sprintf("sts_%s_permission_policy", file))
-			if err != nil {
-				return "", err
-			}
-		} else {
-			policyARN = aws.GetPolicyARN(accountID, accRoleName, path)
+		policyARN, err = aws.GetManagedPolicyARN(input.policies, filename)
+		if err != nil {
+			return err
 		}
 
-		attachRolePolicy := awscb.NewIAMCommandBuilder().
-			SetCommand(awscb.AttachRolePolicy).
-			AddParam(awscb.RoleName, accRoleName).
-			AddParam(awscb.PolicyArn, policyARN).
-			Build()
+		r.Reporter.Debugf("Attaching permission policy to role '%s'", filename)
+		err = r.AWSClient.AttachRolePolicy(accRoleName, policyARN)
+		if err != nil {
+			return err
+		}
+	}
 
-		if managedPolicies {
-			commands = append(commands, createRole, attachRolePolicy)
-		} else {
-			commands = append(commands, createRole, createPolicy, attachRolePolicy)
+	return nil
+}
+
+func (mp managedPoliciesCreator) buildCommands(input rolesCreationInput) (string, error) {
+	commands := []string{}
+	for file, role := range aws.AccountRoles {
+		accRoleName := aws.GetRoleName(input.prefix, role.Name)
+		iamTags := mp.getRoleTags(file, input)
+
+		createRole := buildCreateRoleCommand(accRoleName, file, iamTags, input)
+
+		policyARN, err := aws.GetManagedPolicyARN(input.policies, fmt.Sprintf("sts_%s_permission_policy", file))
+		if err != nil {
+			return "", err
 		}
+
+		attachRolePolicy := buildAttachRolePolicyCommand(accRoleName, policyARN)
+
+		commands = append(commands, createRole, attachRolePolicy)
 	}
 
 	return awscb.JoinCommands(commands), nil
 }
 
-func createRoles(r *rosa.Runtime, prefix, permissionsBoundary, accountID, env string,
-	policies map[string]*cmv1.AWSSTSPolicy, defaultPolicyVersion string,
-	path string, managedPolicies bool) error {
+func (mp managedPoliciesCreator) getRoleTags(roleType string, input rolesCreationInput) map[string]string {
+	tagsList := getBaseRoleTags(roleType, input)
+	tagsList[tags.ManagedPolicies] = "true"
+
+	return tagsList
+}
+
+type unmanagedPoliciesCreator struct{}
```

Comment:
It would be cleaner if the creators are in a separate go file

Reply:
Moved to a new file `creators.go`.